docs/examples/managed/GoogleDemo.ipynb

```diff
@@ -0,0 +1,572 @@
+    ")\n",
+    "from llama_index.vector_stores.google.generativeai import (\n",
+    "    GoogleVectorStore,\n",
+    "    google_service_context,\n",
+    ")\n",
+    "from llama_index import ServiceContext, VectorStoreIndex\n",
+    "from llama_index.llms import PaLM\n",
+    "from llama_index.postprocessor import LLMRerank\n",
+    "from llama_index.query_engine import RetrieverQueryEngine\n",
+    "from llama_index.retrievers import VectorIndexRetriever\n",
+    "\n",
+    "# Set up the query engine with a reranker.\n",
+    "store = GoogleVectorStore.from_corpus(corpus_id=\"some-corpus-id\")\n",
+    "index = VectorStoreIndex.from_vector_store(\n",
+    "    vector_store=store, service_context=google_service_context\n",
+    ")\n",
+    "response_synthesizer = GoogleTextSynthesizer.from_defaults(\n",
+    "    temperature=0.7, answer_style=GenerateAnswerRequest.AnswerStyle.ABSTRACTIVE\n",
+    ")\n",
+    "reranker = LLMRerank(\n",
+    "    top_n=10, service_context=ServiceContext.from_defaults(llm=PaLM())\n",
+    ")\n",
+    "query_engine = RetrieverQueryEngine.from_args(\n",
+    "    retriever=VectorIndexRetriever(\n",
+    "        index=index,\n",
+    "        similarity_top_k=20,\n",
+    "    ),\n",
+    "    response_synthesizer=response_synthesizer,\n",
+    "    node_postprocessors=[reranker],\n",
+    ")\n",
+    "\n",
+    "# Query for better result!\n",
+    "response = query_engine.query(\"What movie should I watch with my family?\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Multi-Query + Google Retriever\n",
+    "\n",
+    "Sometimes, a user's query can be too complex. You may get better retrieval result if you break down the original query into smaller, better focused queries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_index import LLMPredictor\n",
+    "from llama_index.indices.query.query_transform.base import (\n",
+    "    StepDecomposeQueryTransform,\n",
+    ")\n",
+    "from llama_index.prompts.base import PromptTemplate\n",
+    "from llama_index.query_engine.multistep_query_engine import (\n",
+    "    MultiStepQueryEngine,\n",
+    ")\n",
+    "\n",
+    "# Set up the query engine with multi-turn query-rewriter.\n",
+    "store = GoogleVectorStore.from_corpus(corpus_id=\"some-corpus-id\")\n",
+    "index = VectorStoreIndex.from_vector_store(\n",
+    "    vector_store=store, service_context=google_service_context\n",
+    ")\n",
+    "response_synthesizer = GoogleTextSynthesizer.from_defaults(\n",
+    "    temperature=0.7, answer_style=GenerateAnswerRequest.AnswerStyle.ABSTRACTIVE\n",
+    ")\n",
+    "single_step_query_engine = index.as_query_engine(\n",
+    "    response_synthesizer=response_synthesizer\n",
+    ")\n",
+    "step_decompose_transform = StepDecomposeQueryTransform(\n",
+    "    LLMPredictor(llm=PaLM()), verbose=True\n",
```

Comment:
@logan-markewich for LLMPredictor, does our refactoring affect here

Reply:
btw in terms of Gemini, we are integrating with Gemini as well in another pr. should be released on 13 th as well.
if we could switch to Gemini, that would be awesome! But it might require some tests since Gemini would be a new abstraction.
@logan-markewich and I can help change it after Gemini Pr get merged

Reply:
Acknowledged. We can make this small change in another PR when the Gemini PR is out too.